### PR TITLE
Check Salsa and the user store for an email address before sending a verification email

### PR DESCRIPTION
**Pull request by hancush**
## Description

See title.

Handles #7, handles #8.

## Testing instructions

- [x] Update requirement on the salaries staging site.
- [x] Test that login works as expected following the instructions here: https://docs.google.com/document/d/15yjsFpRX8SswXtUeACzzZXsE7qgv4YXLg-FjbfbpYww/edit?usp=sharing

### CODE REVIEW

File: salsa_auth/jinja2/auth_modals.html [hunk 1]
```diff
@@ -35,16 +35,13 @@
                     <p>
                     {% for message in get_messages(request) %}
                         {% if loop.first %}
-                            {% if 'info' in message.tags %}
-                            <h3>You're all set!</h3>
-                            {% elif 'error' in message.tags %}
-                            <h3>Something went wrong.</h3>
-                            {% endif %}
+                            <h3>{{ message|safe }}</h3>
```
Comment (hancush) [inline salsa_auth/jinja2/auth_modals.html:38]: Show the first message as the header.

File: salsa_auth/views.py [hunk 2]
```diff
@@ -47,22 +47,39 @@ class SignUpForm(JSONFormResponseMixin, FormView):
...
+            return redirect('salsa_auth:authenticate')
 
-        messages.add_message(self.request,
-                             messages.INFO,
-                             'Thanks for signing up!',
-                             extra_tags='font-weight-bold')
+        # If the given email is in Salsa, and that email has hard-bounced,
+        # inform the user, and do not send another email to the invalid address.
+        if salsa_client.get_supporter(email, allow_invalid=True):
+            message_title = "Sorry we missed you!"
```
Comment (derekeder) [inline salsa_auth/views.py]: I think having a more informative title here would be helpful so they can take action. Is there any action they can take when their status is 'HARD BOUNCE'? Maybe change this to "We can't verify this email address"

File: salsa_auth/views.py [hunk 3]
```diff
@@ -47,22 +47,39 @@ class SignUpForm(JSONFormResponseMixin, FormView):
...
+        if not user:
+            user = self._make_user(form.cleaned_data)
+
+            self._send_verification_email(user)
+
+            message_title = 'Thanks for signing up!'
+            message_body = 'Please check your email for an activation link.'
+
+        else:
+            message_title = 'You look familiar!'
```
Comment (derekeder) [inline salsa_auth/views.py]: Related to the above comment, change this to "Please verify this email" or something similar

**Review by derekeder — CHANGES_REQUESTED** [review]
Looking good, as far as I can tell. Would be good to test this on a staging site.

Had some comments about the message titles, otherwise looks good

**Review by derekeder — APPROVED** [review]
looks good!

**Pull request merged**